mllib/src/test/scala/org/apache/spark/mllib/linalg/VectorsSuite.scala

```diff
@@ -175,6 +177,33 @@ class VectorsSuite extends FunSuite {
     assert(v.size === x.rows)
   }
 
+  test("sqdist") {
+    val random = new Random(System.nanoTime())
+    for (m <- 1 until 1000 by 10) {
+      val nnz = random.nextInt(m) + 1
```

Comment:
Thanks. I found a bug when nnz is zero in SparseVector vs. DenseVector case. It is fixed in new commit.